service/history/workflow/workflow_task_state_machine.go

```diff
@@ -221,9 +233,6 @@ func (m *workflowTaskStateMachine) AddWorkflowTaskScheduleToStartTimeoutEvent(
 		return nil, m.ms.createInternalServerError(opTag)
 	}
 
-	// clear stickiness whenever workflow task fails
-	m.ms.ClearStickyness()
```

Comment:
this isn't needed here because `ReplicateWorkflowTaskTimedOutEvent(enumspb.TIMEOUT_TYPE_SCHEDULE_TO_START)` below will always call `ClearStickyness` itself